src/main/java/teammates/ui/pagedata/FeedbackSubmissionEditPageData.java

```diff
@@ -249,26 +261,112 @@ private FeedbackSubmissionEditQuestion createQuestion(FeedbackQuestionAttributes
+
+            Map<String, String> commentGiverEmailToNameTable = bundle.getRoster().getEmailToNameTableFromRoster();
+            if (questionAttributes.getQuestionDetails().isFeedbackParticipantCommentsOnResponsesAllowed()) {
+
+                FeedbackResponseCommentRow responseCommentRow = getResponseCommentRowForResponse(questionAttributes,
+                        existingResponse.getId(), commentGiverName, commentRecipientName, commentGiverEmailToNameTable);
+
+                FeedbackResponseCommentRow frcForAdding = buildFeedbackResponseCommentAddFormTemplate(
+                        questionAttributes, existingResponse.getId(), commentGiverName,
+                        commentRecipientName, sessionTimeZone, true);
+
+                responses.add(new FeedbackSubmissionEditResponse(responseIndx,
+                        true, recipientOptionsForQuestion, submissionFormHtml,
+                        existingResponse.getId(), responseCommentRow, frcForAdding));
+
+                responseSubmittedRecipient.add(commentRecipientName);
+            } else {
+                responses.add(new FeedbackSubmissionEditResponse(responseIndx,
+                        true, recipientOptionsForQuestion,
+                        submissionFormHtml, existingResponse.getId(), null, null));
+            }
             responseIndx++;
         }
-
+        int recipientIndxForUnsubmittedResponse = 0;
         while (responseIndx < numOfResponseBoxes) {
             List<String> recipientOptionsForQuestion = getRecipientOptionsForQuestion(questionAttributes.getId(), null);
-            String submissionFormHtml = questionAttributes.getQuestionDetails()
-                                            .getQuestionWithoutExistingResponseSubmissionFormHtml(
-                                                isSessionOpenForSubmission, qnIndx, responseIndx,
-                                                questionAttributes.courseId, numOfResponseBoxes, student);
 
-            responses.add(new FeedbackSubmissionEditResponse(responseIndx, false, recipientOptionsForQuestion,
-                                                             submissionFormHtml, ""));
+            String submissionFormHtml = questionAttributes.getQuestionDetails()
+                    .getQuestionWithoutExistingResponseSubmissionFormHtml(
+                            isSessionOpenForSubmission, qnIndx, responseIndx,
+                            questionAttributes.courseId, numOfResponseBoxes, student);
+
+            if (questionAttributes.getQuestionDetails().isFeedbackParticipantCommentsOnResponsesAllowed()) {
+                List<String> recipientListForUnsubmittedResponse = getRecipientList(responseSubmittedRecipient,
+                        bundle.getSortedRecipientList(questionAttributes.getId()));
+                commentRecipientName = recipientListForUnsubmittedResponse.get(recipientIndxForUnsubmittedResponse);
+                if (isPreview || isModeration) {
```

Comment:
Can we push the logic to `getResponseCommentRowForResponse`?

Reply:
No, `isPreview` and `isModeration` are members of `FeedbackSubmissionEditPageData` class.

Reply:
I don't see problem here as the method is in the same class.